marshal_test.go

```diff
@@ -758,6 +782,27 @@ func TestUnmarshalErrors(t *testing.T) {
 	}
 }
 
+func FuzzUnmarshal(f *testing.F) {
```

Comment:
You can execute this using `go test -v -tags marshal -fuzz '^FuzzUnmarshal$'`. It will not be run when performing normal tests so it should have no negative impact on CI.  If you chose to run this during CI I would recommend adding a timeout to limit how long it runs.

`go test -timeout 30s -v -tags marshal -fuzz '^FuzzUnmarshal$'`

Reply:
Yea, we should run this in CI.

When you did your fuzzing, how long did it typically take to reproduce issues?

Reply:
Most of these were discovered in less than 2 minutes though my original test corpus in my code included more valid packets. You will see better results with more tests because the fuzzer engine will have more to work with.

Reply:
Actually, since `FuzzUnmarshal` includes the test data from `testsUnmarshal` this is likely good enough since `testsUnmarshal` contains many valid SNMP messages.